federation/schema.go

```diff
@@ -177,6 +219,9 @@ func ConvertVersionedSchemas(schemas serviceSchemas) (*SchemaWithFederationInfo,
 	fieldInfos := make(map[*graphql.Field]*FieldInfo)
 	for _, service := range serviceNames {
 		for _, typ := range serviceSchemasByName[service].Schema.Types {
+			if err := validateFederatedObjects(serviceNames, serviceSchemasByName, typ.Name); err != nil {
```

Comment:
I think this call includes some redundant computation since we can just iterate each type once?

Reply:
Im not sure I follow what you mean. I think for every type on every service, we need to check if that type is federated with both the "__federation" field and the "object-service" field